### PR TITLE
qt5-base: fix CVE-2020-24742

### PR DESCRIPTION
**Pull request by mfrw**
Signed-off-by: Muhammad Falak Wani <mwani@microsoft.com>

###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Fix CVE-2020-24742

###### Change Log  

- NA

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

- NA

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2020-24742

###### Test Methodology

- localbuild

### CODE REVIEW

File: SPECS/qt5-qtbase/qt5-qtbase.spec [hunk 1]
```diff
@@ -141,6 +141,7 @@ Patch68: qtbase-everywhere-src-5.11.1-python3.patch
 # https://fedoraproject.org/wiki/Changes/Qt_Wayland_By_Default_On_Gnome
 # https://bugzilla.redhat.com/show_bug.cgi?id=1732129
 Patch80: qtbase-use-wayland-on-gnome.patch
+Patch81: CVE-2020-24742.nopatch
```
Comment (PawelWMS) [inline SPECS/qt5-qtbase/qt5-qtbase.spec:144]: Please add a comment explaining why we nopatch something.

**Review by PawelWMS — CHANGES_REQUESTED** [review]

File: SPECS/qt5-qtbase/qt5-qtbase.spec (hunk 1 above)
Comment (mfrw) [reply SPECS/qt5-qtbase/qt5-qtbase.spec:144]:
```suggestion
# CVE-2020-24742 is not applicable to v5.12.11 (already fixed in this version)
Patch81: CVE-2020-24742.nopatch
```

File: SPECS/qt5-qtbase/qt5-qtbase.spec [hunk 2]
```diff
@@ -141,6 +141,7 @@ Patch68: qtbase-everywhere-src-5.11.1-python3.patch
 # https://fedoraproject.org/wiki/Changes/Qt_Wayland_By_Default_On_Gnome
 # https://bugzilla.redhat.com/show_bug.cgi?id=1732129
 Patch80: qtbase-use-wayland-on-gnome.patch
+# CVE-2020-24742 is not applicable to v5.12.11 (already fixed in this version)
```
Comment (PawelWMS) [inline SPECS/qt5-qtbase/qt5-qtbase.spec:144]: Ah, I see now. This one's a bit tricky, because the CVE fix's "coverage" is not continuous according to the [official messages](https://codereview.qt-project.org/c/qt/qtbase/+/280730): versions 5.5, 5.9, and 5.12 are fixed, the issue is not present in 5.14, so 5.13 still potentially remains as being vulnerable. Considering that, and the fact, that we've already asked NIST to update they CPE information to be more detailed, I think we should simply do nothing and wait for that NIST update, which will make this CVE disappear from our radar as well.

In short: this is a no-op for us, we just need to wait for NIST to update their database.

**Review by PawelWMS — CHANGES_REQUESTED** [review]

**Comment by mfrw** [comment]
Lets hold on merging this.

**Comment by jslobodzian** [comment]
Closing the PR since our version of the package is not impacted by this CVE.
